airbyte-integrations/connectors/source-close-com/source_close_com/source.py

```diff
@@ -0,0 +1,706 @@
+    """
+
+    _type = "emailthread"
+
+
+class EmailActivities(CloseComActivitiesMixin):
+    """
+    Get email activities on a specific date
+    API Docs: https://developer.close.com/#activities-list-or-filter-all-email-activities
+    """
+
+    _type = "email"
+
+
+class SmsActivities(CloseComActivitiesMixin):
+    """
+    Get SMS activities on a specific date
+    API Docs: https://developer.close.com/#activities-list-or-filter-all-sms-activities
+    """
+
+    _type = "sms"
+
+
+class CallActivities(CloseComActivitiesMixin):
+    """
+    Get call activities on a specific date
+    API Docs: https://developer.close.com/#activities-list-or-filter-all-call-activities
+    """
+
+    _type = "call"
+
+
+class MeetingActivities(CloseComActivitiesMixin):
+    """
+    Get meeting activities on a specific date
+    API Docs: https://developer.close.com/#activities-list-or-filter-all-meeting-activities
+    """
+
+    _type = "meeting"
+
+
+class LeadStatusChangeActivities(CloseComActivitiesMixin):
+    """
+    Get lead status change activities on a specific date
+    API Docs: https://developer.close.com/#activities-list-or-filter-all-leadstatuschange-activities
+    """
+
+    _type = "status_change/lead"
+
+
+class OpportunityStatusChangeActivities(CloseComActivitiesMixin):
+    """
+    Get opportunity status change activities on a specific date
+    API Docs: https://developer.close.com/#activities-list-or-filter-all-opportunitystatuschange-activities
+    """
+
+    _type = "status_change/opportunity"
+
+
+class TaskCompletedActivities(CloseComActivitiesMixin):
+    """
+    Get task completed activities on a specific date
+    API Docs: https://developer.close.com/#activities-list-or-filter-all-taskcompleted-activities
+    """
+
+    _type = "task_completed"
+
+
+class Events(IncrementalCloseComStream):
+    """
+    Get events on a specific date
+    API Docs: https://developer.close.com/#event-log-retrieve-a-list-of-events
+    """
+
+    number_of_items_per_page = 50
+
+    def path(self, **kwargs) -> str:
+        return "event"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if stream_state.get(self.cursor_field):
+            params["date_updated__gt"] = stream_state.get(self.cursor_field)
+        return params
+
+
+class Leads(IncrementalCloseComStream):
+    """
+    Get leads on a specific date
+    API Docs: https://developer.close.com/#leads
+    """
+
+    def path(self, **kwargs) -> str:
+        return "lead"
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if stream_state.get(self.cursor_field):
+            params["query"] = f"date_updated > {stream_state.get(self.cursor_field)}"
```

Comment:
this is a general point on the incremental filters...

if we're changing both the filter and the _skip value in each request, is it possible we're missing data? e.g. 

1. state=X
2. we read 100 records 
3. update state to Y
4. we now send request for next 100 records with _skip=100 and filtering for records with updated_at > Y

Would this not skip 100 records where updated_at > Y that we haven't seen yet (since our first 100 records had updated_at <= Y)?

Reply:
Thanks! Understand the problem. So, I've added ordering for streams with incremental sync. And for activities and events remove `state_checkpoint_interval` due to the impossibility of ordering data ascending.